src/ResourceManager/Compute/Commands.Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetUpdateMethod.cs

```diff
@@ -110,4 +113,877 @@ protected PSArgument[] CreateVirtualMachineScaleSetUpdateParameters()
+    [OutputType(typeof(PSVirtualMachineScaleSet))]
+    public partial class UpdateAzureRmVmss : ComputeAutomationBaseCmdlet
+    {
+        protected override void ProcessRecord()
+        {
+            ExecuteClientAction(() =>
+            {
+                if (ShouldProcess(this.VMScaleSetName, VerbsData.Update))
+                {
+
+                    string resourceGroupName = this.ResourceGroupName;
+                    string vmScaleSetName = this.VMScaleSetName;
+                    BuildPatchObject();
+                    VirtualMachineScaleSetUpdate parametersupdate = this.VirtualMachineScaleSetUpdate;
+                    VirtualMachineScaleSet parameters = new VirtualMachineScaleSet();
+                    ComputeAutomationAutoMapperProfile.Mapper.Map<PSVirtualMachineScaleSet, VirtualMachineScaleSet>(this.VirtualMachineScaleSet, parameters);
+
+                    var result = (this.VirtualMachineScaleSetUpdate == null)
+                                 ? VirtualMachineScaleSetsClient.CreateOrUpdate(resourceGroupName, vmScaleSetName, parameters)
+                                 : VirtualMachineScaleSetsClient.Update(resourceGroupName, vmScaleSetName, parametersupdate);
+                    var psObject = new PSVirtualMachineScaleSet();
+                    ComputeAutomationAutoMapperProfile.Mapper.Map<VirtualMachineScaleSet, PSVirtualMachineScaleSet>(result, psObject);
+                    WriteObject(psObject);
+                }
+            });
+        }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ValueFromPipeline = false)]
+        [Parameter(
+            ParameterSetName = "FriendMethod",
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ValueFromPipeline = false)]
+        [AllowNull]
+        [ResourceManager.Common.ArgumentCompleters.ResourceGroupCompleter()]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Position = 2,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ValueFromPipeline = false)]
+        [Parameter(
+            ParameterSetName = "FriendMethod",
+            Position = 2,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ValueFromPipeline = false)]
+        [Alias("Name")]
+        [AllowNull]
+        public string VMScaleSetName { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Position = 3,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = false,
+            ValueFromPipeline = true)]
+        [AllowNull]
+        public VirtualMachineScaleSetUpdate VirtualMachineScaleSetUpdate { get; set; }
+
```

Comment:
These parameter sets are incorrect.  

The intention shoudl be to 
1. Allow identifying the VMSS by resource group + resource name
2. Allow piping in a VMSS via the pipeline and updating the parameters.

In both of these cases, ideally you would allow either patching individual properties (if they provide additional parameters), or simply writing the VMSS object.

Firther, It is not clear how the user creates a VisrtualMachienScaleSetUpdate parameter,a s this doesn't appear to be the ouput of any other cmdlet.